// @elastic/eui/optimize/es/components/timeline/timeline_item.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";

/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */
import React from 'react';
import { useEuiTheme } from '../../services';
import { EuiTimelineItemEvent } from './timeline_item_event';
import { EuiTimelineItemIcon } from './timeline_item_icon';
import { euiTimelineItemStyles } from './timeline_item.styles';
import { jsx as ___EmotionJSX } from "@emotion/react";
export var VERTICAL_ALIGN = ['top', 'center'];
export var EuiTimelineItem = function EuiTimelineItem(_ref) {
  var children = _ref.children,
      _ref$verticalAlign = _ref.verticalAlign,
      verticalAlign = _ref$verticalAlign === void 0 ? 'center' : _ref$verticalAlign,
      icon = _ref.icon,
      iconAriaLabel = _ref.iconAriaLabel,
      className = _ref.className,
      _ref$component = _ref.component,
      component = _ref$component === void 0 ? 'li' : _ref$component,
      rest = _objectWithoutProperties(_ref, ["children", "verticalAlign", "icon", "iconAriaLabel", "className", "component"]);

  var euiTheme = useEuiTheme();
  var styles = euiTimelineItemStyles(euiTheme);
  var cssStyles = [styles.euiTimelineItem];
  var Element = component;
  return ___EmotionJSX(Element, _extends({
    css: cssStyles
  }, rest), ___EmotionJSX(EuiTimelineItemIcon, {
    icon: icon,
    iconAriaLabel: iconAriaLabel,
    verticalAlign: verticalAlign
  }), ___EmotionJSX(EuiTimelineItemEvent, {
    verticalAlign: verticalAlign
  }, children));
};
