// @elastic/eui/test-env/components/timeline/timeline.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EuiTimeline = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _timeline_item = require("./timeline_item");

var _react2 = require("@emotion/react");

/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */
var EuiTimeline = function EuiTimeline(_ref) {
  var className = _ref.className,
      _ref$items = _ref.items,
      items = _ref$items === void 0 ? [] : _ref$items,
      children = _ref.children,
      rest = (0, _objectWithoutProperties2.default)(_ref, ["className", "items", "children"]);
  var classes = (0, _classnames.default)('euiTimeline', className);
  return (// eslint-disable-next-line jsx-a11y/no-redundant-roles
    (0, _react2.jsx)("ol", (0, _extends2.default)({
      className: classes,
      role: "list"
    }, rest), items.map(function (item, index) {
      return (0, _react2.jsx)(_timeline_item.EuiTimelineItem, (0, _extends2.default)({
        key: index
      }, item));
    }), children)
  );
};

exports.EuiTimeline = EuiTimeline;
EuiTimeline.propTypes = {
  /**
     * List of timeline items to render. See #EuiTimelineItem
     */
  items: _propTypes.default.arrayOf(_propTypes.default.shape({
    /**
       * Vertical alignment of the event with the icon
       */
    verticalAlign: _propTypes.default.any,

    /**
       * Sets the HTML element for `EuiTimelineItem`.
       * By default, the element renders as a `<li/>`.
       * Only change the HTML element when it is not wrapped in a `EuiTimeline` that renders as a `<ol/>`.
       */
    component: _propTypes.default.any,
    className: _propTypes.default.string,
    "aria-label": _propTypes.default.string,
    "data-test-subj": _propTypes.default.string
  }).isRequired),
  className: _propTypes.default.string,
  "aria-label": _propTypes.default.string,
  "data-test-subj": _propTypes.default.string
};
