// @elastic/eui/test-env/themes/amsterdam/global_styling/variables/_states.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.focus = void 0;

var _utils = require("../../../../services/theme/utils");

var _color = require("../../../../services/color");

var _size = require("../../../../global_styling/functions/size");

/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

/**
 * NOTE: These were quick conversions of their Sass counterparts.
 *       They have yet to be used/tested.
 */
var focus = {
  color: 'currentColor',
  transparency: {
    LIGHT: 0.9,
    DARK: 0.7
  },
  backgroundColor: (0, _utils.computed)(function (_ref) {
    var colors = _ref.colors,
        focus = _ref.focus;
    return (0, _color.transparentize)(colors.primary, focus.transparency);
  }),
  // Sizing
  widthLarge: (0, _utils.computed)((0, _size.sizeToPixel)(0.25)),
  width: (0, _utils.computed)((0, _size.sizeToPixel)(0.125)),
  // Outline
  outline: {
    outline: (0, _utils.computed)(function (_ref2) {
      var focus = _ref2.focus;
      return "".concat(focus.width, " solid ").concat(focus.color);
    })
  }
};
exports.focus = focus;
